lib/utils/discover.ts

```diff
@@ -7,34 +8,41 @@ interface LocalBalenaOsDevice {
 	port: number;
 }
 
-// Although we only check for 'balena-ssh', we know, implicitly, that balenaOS
-// devices come with 'rsync' installed that can be used over SSH.
-const avahiBalenaSshTag = 'resin-ssh';
+const avahiBalenaSshConfig = {
+	type: 'ssh',
+	name: '_resin-device._sub',
+	protocol: 'tcp' as const,
+};
+
+const avahiBalenaSshSubtype = 'resin-device';
 
 export async function discoverLocalBalenaOsDevices(
 	timeout = 4000,
 ): Promise<LocalBalenaOsDevice[]> {
-	const availableServices = await enumerateServices();
-	const serviceDefinitions = Array.from(availableServices)
-		.filter((s) => Array.from(s.tags).includes(avahiBalenaSshTag))
-		.map((s) => s.service);
-
-	if (serviceDefinitions.length === 0) {
-		throw new Error(
-			`Could not find any available '${avahiBalenaSshTag}' services`,
+	const bonjour = new Bonjour({}, async (err: string | Error) => {
+		await (await import('../errors')).handleError(err);
+	});
+	const services = await new Promise<Service[]>((resolve) => {
+		const services: Service[] = [];
+		const browser = bonjour.find(avahiBalenaSshConfig, (service) =>
+			services.push(service),
 		);
-	}
+		setTimeout(() => {
+			browser.stop();
+			resolve(services);
+		}, timeout);
+	});
```

Comment:
You're not using Bonjour besides for getting the services, so creating and destroying the instance within that promise, so that its lifetime is fully contained, might be a good idea.

Other than that, the PR looks great to me and feel free to merge when you're ready.

Reply:
Ah! yes I understand what you mean now. Yes, this is a great suggestion.